dds/DCPS/security/framework/HandleRegistry.cpp

```diff
@@ -63,12 +75,12 @@ HandleRegistry::get_local_datareader_crypto_handle(const DCPS::RepoId& id) const
 DDS::Security::EndpointSecurityAttributes
 HandleRegistry::get_local_datareader_security_attributes(const DCPS::RepoId& id) const
 {
-  ACE_GUARD_RETURN(ACE_Thread_Mutex, guard, mutex_, DDS::Security::EndpointSecurityAttributes());
+  ACE_GUARD_RETURN(ACE_Thread_Mutex, guard, mutex_, default_endpoint_security_attributes_);
   DatareaderCryptoHandleMap::const_iterator pos = local_datareader_crypto_handles_.find(id);
   if (pos != local_datareader_crypto_handles_.end()) {
     return pos->second.second;
   }
-  return DDS::Security::EndpointSecurityAttributes();
+  return default_endpoint_security_attributes_;
 }
```

Comment:
Can this method return const DDS::Security::EndpointSecurityAttributes or const DDS::Security::EndpointSecurityAttributes& ?

Reply:
We can do a reference.

Reply:
Would it be also applicable for `get_local_datareader_crypto_handle` above?